.github/actions/setup-binary/action.yml

```diff
@@ -0,0 +1,135 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Setup Binary'
+
+description: |-
+  Use this action to install and cache a binary for use within a GitHub workflow.
+
+inputs:
+  name:
+    description: 'The name of the binary.'
+    required: true
+  download_url:
+    description: 'The download URL for the binary.'
+    required: true
+  install_dir:
+    description: 'The directory to install the binary (e.g. install/dir).'
+    default: '.'
+    required: false
+  binary_dir:
+    description: 'The directory to the binary. This is useful when the binary is located in a sub-path of the downloaded archive (e.g. install/dir/sub/path).'
+    required: false
```

Comment:
`name`, `install_dir`, and `binary_dir` are confusing names to me. I think I'd expect:

- `install_path` - path on disk in which to install the binary (defaults to XXX). This includes the folder and binary name (e.g. /path/to/my/binary). This action does not modify `$PATH`; if you want the binary to be accessible, make sure the parent folder is in `$PATH`.

- `archive_binary_path` - path within the extracted archive in which the binary is located. This is useful when the binary is in a sub-directory of the extracted archive (e.g. bin/). This value is relative to the root of the extracted archive.

Reply:
SGTM, we dont need archive_binary_path is we arent setting the path. The user can just set the path to the subdir on the outside of the action.